ha/fault_tolerance/fault_tolerance_driver.py

```diff
@@ -33,33 +31,45 @@ class FaultTolerance:
     Module responsible for consuming messages from message bus,
     further analyzes that event and publishes it if required
     """
-    def __init__(self, wait_time=10):
-        """Init method"""
-        self._wait_time = wait_time
+    def __init__(self):
+        """
+        Init method
+        Create monitor objects and Sets the callbacks to sigterm
+        """
+        signal.signal(signal.SIGTERM, self.set_sigterm)
         ConfigManager.init("fault_tolerance")
         self.node_fault_monitor = NodeFaultMonitor()
         self.cluster_stop_monitor = ClusterStopMonitor()
 
+    def set_sigterm(self, signum, frame):
+        """
+        Callback function to receive a signal
+        """
+        Log.info(f"Received SIGTERM {signum}")
+        Log.debug(f"Received signal: {signum} during execution of frame: {frame}")
+        Log.info("Stopping the Fault Tolerance Monitor...")
+        self.node_fault_monitor.stop(flush=True)
+        self.cluster_stop_monitor.stop(flush=True)
+
     def start(self):
         """
         start the threads
         """
+        Log.info("Starting all daemon threads of Fault Tolerance Monitor...")
         self.node_fault_monitor.start()
         self.cluster_stop_monitor.start()
+        Log.info(f"Fault Tolerance Monitor with PID {os.getpid()} started successfully.")
 
-    def poll(self):
+    def wait_for_exit(self):
         """
-        wait method for receiving events
+        join and wait for monitor threads to exit
         """
-        Log.debug("FaultTolerance poll")
-        try:
-            while True:
-                time.sleep(self._wait_time)
-        except Exception as exe:
-            raise(f'Oops, some issue in the fault tolerance_driver: {exe}')
+        self.node_fault_monitor.join()
```

Comment:
What happens to threads if there is an exception ? 
This is a generic query for the modified functionality.
Is the behavior same as before in case Exceptions are received in a thread ?

Reply:
main thread which is joining will wait till all thread joins.
no matter who gets the exception and stops.
the thread will by default print stack-trace but the mail thread which is waiting will not get notified. 

Well, catch! this needs handle but look like the previous while loop was also not getting notify event if it was trying to check exception.

Needs to handle this

Reply:
new issue regarding this: https://jts.seagate.com/browse/EOS-27291